mvpa2/datasets/mri.py

```diff
@@ -481,3 +481,77 @@ def strip_nibabel(ds):
         # when conversion fails, we need to kill the remains
         del ds.a['imghdr']
     return ds
+
+
+def preprocessed_fmri_dataset(
+        fmri_fname, preproc_img=None, preproc_ds=None, add_sa=None,
+        **kwargs):
+    """
+    Convenience function to load and preprocess an fMRI dataset.
+
+    This function wraps ``fmri_dataset()`` and supports all its arguments.
+    However, there are a few limitations (see below). Specifically, this
+    function does three things.
+
+    1. Provide an interface for pre-processing in volumetric image space.
+    2. Provide an interface for time-series pre-processing.
+    3. Convenience functionality to add sample attributes to the dataset.
+
+    First the input image is loaded with NiBabel and any desired volumetric
+    pre-processing is performed. This can be used to apply, for example,
+    spatial smoothing. Next a PyMVPA dataset is created via a call to
+    ``fmri_dataset()``. Afterwards, additional sample attributes are assigned.
+    Lastly, the resulting (flattened and masked) dataset is subjected to
+    another pre-processing step that can be used to , for example, apply
+    temporal filtering.
+
+    Limitations with respect to ``fmri_dataset()`` exist insofar as this
+    function expects a single (4D) image to be specified, while calling
+    ``fmri_dataset()`` directly allows for specifying lists of individual
+    image (file names).
+
```

Comment:
Are u sure this additional functionality shouldn't get absorbed into fmri_dataset?
Or alternatively, it could drop fmri_dataset specificity- provide that func as an argument and then could be used with any dataset constructor or factory